mux.go

```diff
@@ -344,6 +354,11 @@ type RouteMatch struct {
 	Route   *Route
 	Handler http.Handler
 	Vars    map[string]string
+
+	// MatchErr is set to appropriate matching error
+	// It is set to ErrMethodMismatch if there is a mismatch in
+	// the request method and route method
+	MatchErr error
```

Comment:
I agree that router should have MethodNotAllowedHandler. We can default to local `methodNotAllowedHandler` if it is not set, same as we do with `http.NotFoundHandler`. Should we make those change in a current fix #271 is the question?